api/src/api/cookie/services/cookie.js

```diff
@@ -0,0 +1,21 @@
+'use strict';
+
+/**
+ * cookie service
+ */
+
+const { createCoreService } = require('@strapi/strapi').factories;
+
+module.exports = createCoreService('api::cookie.cookie', ({ strapi }) => ({
+  async find(params) {
+    const { pagination } = await super.find(params);
+    const results = await strapi.entityService.findMany(
```

Comment:
Not sure why this method was used here, I would suggest using a route middleware instead if you just want to set the default populate, customizing the controller seems a bit much. Paul has an example blog post that Kellen transcribed from a loom I made that covers this as well.

Reply:
Basically you are doubling the total DB queries being made but one of them is thrown away.

Reply:
I've not tested that plugin but we should evaluate these and see if we can optimize it. I'd really like food advisor to be a way we can express best practices as an example.

Reply:
Should I remove the plugin?

Reply:
@Mcastres - managed to look at this? So that I can approve the PR, sorry for taking to look to review :(

Reply:
@derrickmehaffy should I removed the plugin so?

Reply:
I don't think we need to remove the plugin (tbh i don't know really what it does) but this example doesn't make sense from the controller level. It's fine to populate if that's what we need to do but it should be done in a route middleware, not the controller.